### DISCUSSION TITLE
How does the installer enable thumbnail generation on Windows ?

### DISCUSSION THREAD

**Discussion by ompadu** [sumatrapdfreader/sumatrapdf#4336] [Q&A]
I am installing SumatraPDF via the scoop command line installer, which makes use of the portable version, thus i have no access to the checkbox in the graphical installer. 

I'd like to independently replicate that process. How does the installer do it? 

Thank you and best wishes

**Comment by GitHubRulesOK** [comment]
The installer is not portable as iFilter and previewer have to be installed into explorer, thus you use the installer version with the full ability to Authorise UAC (hence all the moaning it needs admin elevation to attack the registry during install and uninstall).

Basically the portable (standalone) version does not have any filters so no need for elevation and thus unable to install them, hence its portable-ish (less so now than before!)

**Reply by ompadu** [reply]
How are IFilter and previewer installed into explorer? Dlls copied somewhere and some registry keys? Where can i find out exactly what the installer does?

**Reply by GitHubRulesOK** [reply]
the steps needed will be in the source code for the installer but  you dont need to do that simply run the installer it will do everything necessary since it is a hybrid "ONE.EXE" 

It is the dlls and the elevator and the registry editor and recognised as signed  by windows as both uninstaller and installer so as to disable and reenable explorer filters.

 You will be causing untold user support problems if you attempt to emulate in any DIY method

**Reply by GitHubRulesOK** [reply]
Let me point out a facet you may be overlooking.

The core code of The SumatraPDF application depends on MuPDF which is a Licensed product. 

Thus SumatraPDF is licensed in a very simple fashion. it is

Free Open Source solely as long as it is distributed as released 
Thus you may use the 3 different instaler.exe(s) as long as each is one.exe UNMODIFIED

So simply use the installer as designed and it will be licensed for any user/usage

```
-with-filter : install search filter
-with-preview : install shell preview for PDF files
```

So using the provided EXEs, there are already more than 36 ways to install and thus the uninstaller needs to keep track and know how to reverse them. Adding more complexity is pointless
